File: checkout/src/main/java/com/payoneer/checkout/util/Event.java
/*
 * Copyright (c) 2022 Payoneer Germany GmbH
 * https://www.payoneer.com
 *
 * This file is open source and available under the MIT license.
 * See the LICENSE file for more information.
 */

package com.payoneer.checkout.util;

/**
 * Generic event class to be used with LiveData
 */
public class Event {

    private boolean hasBeenHandled;

    public Event getIfNotHandled() {
        if (hasBeenHandled) {
            return null;
        }
        hasBeenHandled = true;
        return this;
    }
}

File: example-shop/src/main/java/com/payoneer/checkout/exampleshop/util/Event.kt
/*
 *
 *  * Copyright (c) 2022 Payoneer Germany GmbH
 *  * https://www.payoneer.com
 *  *
 *  * This file is open source and available under the MIT license.
 *  * See the LICENSE file for more information.
 *  *
 */

package com.payoneer.checkout.exampleshop.util

/**
 * Used as a wrapper an event that has no data
 */
open class Event {

    private var hasBeenHandled = false

    fun getIfNotHandled(): Event? {
        return if (hasBeenHandled)
            null
        else {
            hasBeenHandled = true
            this
        }
    }

}